### PR TITLE
AUTO-71 / 22.12 / Adding API test cases to test system dataset 

### PR DESCRIPTION
**Pull request by ericbsd**

### CODE REVIEW

File: tests/api2/test_007_systemdataset.py [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 240)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+    pool_data['encrypted'] = job_status['results']['result']
+
+    results = GET("/systemdataset/")
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+    assert results.json()['pool'] == 'encrypted', results.text
+    assert results.json()['basename'] == 'encrypted/.system', results.textZ
```
Comment (yocalebo) [inline tests/api2/test_007_systemdataset.py:92]:
```suggestion
    assert results.json()['basename'] == 'encrypted/.system', results.text
```

File: tests/api2/test_007_systemdataset.py [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+    assert journald_log['result'] is True, str(journald_log)
+    logs_data['journald_log_1'] = journald_log['output'].splitlines()[-1]
+
+    cmd = "cat /var/log/syslog"
+    syslog = SSH_TEST(cmd, user, password, ip)
+    assert syslog['result'] is True, str(syslog)
+    logs_data['syslog_1'] = syslog['output'].splitlines()[-1]
+
+
+def test_03_verify_the_first_pool_created_with_encrypted_root_dataset_become_the_system_dataset(request, pool_data):
```
Comment (yocalebo) [inline tests/api2/test_007_systemdataset.py:66]:
```suggestion
def test_03_verify_sysds_is_moved_after_first_pool_is_created(request, pool_data):
```

File: tests/api2/test_007_systemdataset.py [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+
+    results = GET("/systemdataset/")
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+    assert results.json()['pool'] == 'encrypted', results.text
+    assert results.json()['basename'] == 'encrypted/.system', results.text
+
+
+def test_05_verify_logs_collection_still_work_after_enrcripted_dataset_is_moved_to_system_dataset(logs_data):
```
Comment (yocalebo) [inline tests/api2/test_007_systemdataset.py:115]:
```suggestion
def test_05_verify_logs_after_sysds_is_moved_to_a_passphrase_encrypted_pool(logs_data):
```

File: tests/api2/test_007_systemdataset.py [hunk 4]
```diff
@@ -0,0 +1,409 @@
...
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+
+    results = GET("/systemdataset/")
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+    assert results.json()['pool'] == 'boot-pool', results.text
+    assert results.json()['basename'] == 'boot-pool/.system', results.text
+
+
+def test_07_verify_logs_collection_still_work_after_enrcripted_dataset_is_deleted_still_work(logs_data):
```
Comment (yocalebo) [inline tests/api2/test_007_systemdataset.py:157]:
```suggestion
def test_07_verify_logs_after_passphrase_encrypted_pool_is_deleted(logs_data):
```

File: tests/api2/test_007_systemdataset.py [hunk 5]
```diff
@@ -0,0 +1,409 @@
...
+    pool_data['encrypted'] = job_status['results']['result']
+
+    results = GET("/systemdataset/")
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+    assert results.json()['pool'] == 'encrypted', results.text
+    assert results.json()['basename'] == 'encrypted/.system', results.textZ
+
+
+def test_04_verify_the_system_dataset_can_move_to_an_encrypted_root_dataset(request):
```
Comment (yocalebo) [inline tests/api2/test_007_systemdataset.py:95]:
```suggestion
def test_04_verify_sysds_can_move_to_a_passphrase_encrypted_pool(request):
```

File: tests/api2/test_007_systemdataset.py [hunk 6]
```diff
@@ -0,0 +1,409 @@
...
+def logs_data():
+    return {}
+
+
+@pytest.fixture(scope='module')
+def reporing_data():
+    return {}
+
+
+def test_01_verify_system_dataset_is_set_to_boot_pool():
```
Comment (yocalebo) [inline tests/api2/test_007_systemdataset.py:41]:
```suggestion
def test_01_verify_sysds_is_set_to_boot_pool():
```

File: tests/api2/test_007_systemdataset.py [hunk 7]
```diff
@@ -0,0 +1,409 @@
...
+
+    cmd = "cat /var/log/syslog"
+    syslog = SSH_TEST(cmd, user, password, ip)
+    assert syslog['result'] is True, str(syslog)
+    logs_data['syslog_2'] = syslog['output'].splitlines()[-1]
+    assert logs_data['syslog_1'] in syslog['output'], str(syslog['output'])
+    assert logs_data['syslog_1'] != logs_data['syslog_2']
+
+
+def test_06_delete_the_encrypted_pool_and_verify_the_system_dataset(request, pool_data):
```
Comment (yocalebo) [inline tests/api2/test_007_systemdataset.py:138]:
```suggestion
def test_06_verify_sysds_after_passphrase_encrypted_pool_is_deleted(request, pool_data):
```

File: tests/api2/test_007_systemdataset.py [hunk 8]
```diff
@@ -0,0 +1,409 @@
...
+    cmd = "cat /var/log/syslog"
+    syslog = SSH_TEST(cmd, user, password, ip)
+    assert syslog['result'] is True, str(syslog)
+    logs_data['syslog_3'] = syslog['output'].splitlines()[-1]
+    assert logs_data['syslog_2'] in syslog['output'], str(syslog['output'])
+    assert logs_data['syslog_2'] != logs_data['syslog_3']
+
+
+@pytest.mark.dependency(name="first_pool")
+def test_08_creating_a_first_pool_and_verify_system_dataset_move_to_the_new_pool(request, pool_data):
```
Comment (yocalebo) [inline tests/api2/test_007_systemdataset.py:181]:
```suggestion
def test_08_verify_sysds_moves_to_first_created_zpool(request, pool_data):
```

File: tests/api2/test_007_systemdataset.py [hunk 9]
```diff
@@ -0,0 +1,409 @@
...
+
+    results = GET("/systemdataset/")
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+    assert results.json()['pool'] == 'first_pool', results.text
+    assert results.json()['basename'] == 'first_pool/.system', results.text
+
+
+@pytest.mark.dependency(name="second_pool")
+def test_09_creating_a_second_pool_and_verify_system_dataset_does_not_move_to_the_new_pool(request, pool_data):
```
Comment (yocalebo) [inline tests/api2/test_007_systemdataset.py:208]:
```suggestion
def test_09_verify_sysds_does_not_move_after_second_pool_is_created(request, pool_data):
```

File: tests/api2/test_007_systemdataset.py [hunk 10]
```diff
@@ -0,0 +1,409 @@
...
+    pool_data['second_pool'] = job_status['results']['result']
+
+    results = GET("/systemdataset/")
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+    assert results.json()['pool'] == 'first_pool', results.text
+    assert results.json()['basename'] == 'first_pool/.system', results.text
+
+
+def test_10_verify_changing_a_system_dataset_is_impossible_while_AD_is_running(request):
```
Comment (yocalebo) [inline tests/api2/test_007_systemdataset.py:235]:
```suggestion
def test_10_verify_changes_to_sysds_are_forbidden_while_AD_is_running(request):
```

File: tests/api2/test_007_systemdataset.py [hunk 11]
```diff
@@ -0,0 +1,409 @@
...
+    results = POST("/activedirectory/leave/", leave_payload)
+    assert results.status_code == 200, results.text
+    job_status = wait_on_job(results.json(), 180)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+
+    results = PUT("/network/configuration/", {"nameserver1": nameserver1})
+    assert results.status_code == 200, results.text
+
+
+def test_11_get_logs_before_moving_the_system_dataset_to_the_second_pool(logs_data):
```
Comment (yocalebo) [inline tests/api2/test_007_systemdataset.py:292]:
```suggestion
def test_11_get_logs_before_moving_the_sysds_to_the_second_pool(logs_data):
```

File: tests/api2/test_007_systemdataset.py [hunk 12]
```diff
@@ -0,0 +1,409 @@
...
+    assert journald_log['result'] is True, str(journald_log)
+    logs_data['journald_log_4'] = journald_log['output'].splitlines()[-1]
+
+    cmd = "cat /var/log/syslog"
+    syslog = SSH_TEST(cmd, user, password, ip)
+    assert syslog['result'] is True, str(syslog)
+    logs_data['syslog_4'] = syslog['output'].splitlines()[-1]
+
+
+def test_12_a_system_dataset_can_be_moved_to_second_pool_root_dataset(request):
```
Comment (yocalebo) [inline tests/api2/test_007_systemdataset.py:309]:
```suggestion
def test_12_move_sysds_to_second_pool(request):
```

File: tests/api2/test_007_systemdataset.py [hunk 13]
```diff
@@ -0,0 +1,409 @@
...
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+
+    results = GET("/systemdataset/")
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+    assert results.json()['pool'] == 'second_pool', results.text
+    assert results.json()['basename'] == 'second_pool/.system', results.text
+
+
+def test_13_verify_logs_collection_still_work_after_moving_the_system_dataset_to_the_second_pool(logs_data):
```
Comment (yocalebo) [inline tests/api2/test_007_systemdataset.py:324]:
```suggestion
def test_13_verify_logs_after_sysds_is_moved_to_second_pool(logs_data):
```

File: tests/api2/test_007_systemdataset.py [hunk 14]
```diff
@@ -0,0 +1,409 @@
...
+
+    cmd = "cat /var/log/syslog"
+    syslog = SSH_TEST(cmd, user, password, ip)
+    assert syslog['result'] is True, str(syslog)
+    logs_data['syslog_5'] = syslog['output'].splitlines()[-1]
+    assert logs_data['syslog_4'] in syslog['output'], str(syslog['output'])
+    assert logs_data['syslog_4'] != logs_data['syslog_5']
+
+
+def test_14_system_dataset_can_be_moved_to_another_pool_successfully_when_all_services_running(request):
```
Comment (yocalebo) [inline tests/api2/test_007_systemdataset.py:347]:
```suggestion
def test_14_verify_sysds_can_be_moved_while_services_are_running(request):
```

File: tests/api2/test_007_systemdataset.py [hunk 15]
```diff
@@ -0,0 +1,409 @@
...
+    assert results.json()['pool'] == 'first_pool', results.text
+    assert results.json()['basename'] == 'first_pool/.system', results.text
+
+    for service in services_list:
+        if service != 'ssh':
+            results = POST("/service/stop/", {"service": service})
+            assert results.status_code == 200, results.text
+
+
+def test_15_delete_the_second_pool_and_verify_the_system_dataset(request, pool_data):
```
Comment (yocalebo) [inline tests/api2/test_007_systemdataset.py:374]:
```suggestion
def test_15_delete_second_pool_and_verify_sysds_is_moved_to_first_pool(request, pool_data):
```

File: tests/api2/test_007_systemdataset.py [hunk 16]
```diff
@@ -0,0 +1,409 @@
...
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+
+    results = GET("/systemdataset/")
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+    assert results.json()['pool'] == 'first_pool', results.text
+    assert results.json()['basename'] == 'first_pool/.system', results.text
+
+
+def test_16_delete_the_firs_pool_and_verify_the_system_dataset_moved_to_the_boot_pool(request, pool_data):
```
Comment (yocalebo) [inline tests/api2/test_007_systemdataset.py:393]:
```suggestion
def test_16_delete_first_pool_and_verify_sysds_moved_to_the_boot_pool(request, pool_data):
```

**Review by yocalebo — CHANGES_REQUESTED** [review]

**Comment by ericbsd** [comment]
> Committed some of the changes but not all of them?

I missed some, but they should all be there now.

**Review by yocalebo — APPROVED** [review]

**Pull request merged**